### ISSUE TITLE
Import error JsZip

### ISSUE THREAD

**Issue by risalfajar** [PepsRyuu/nollup#215]
I'm using Jszip for my project:
```
    import JSZip from 'jszip';
    JSZip.loadAsync(file).then(value => console.log(value))
```

But when using Nollup, it throws an error on runtime
`Uncaught TypeError: JSZip.loadAsync is not a function`

This does not happen when I'm using Rollup, so I think this is the problem with my configuration or something.

.nolluprc.js
```
import appConfig from './appConfig'
import {relative} from 'path'

const { assetsDir, distDir, buildDir, template } = appConfig

module.exports = {
  hot: true,
  contentBase: assetsDir,
  publicPath: relative(distDir, buildDir),
  historyApiFallback: relative(assetsDir, template),
  port: 5002
}
```

rollup.config.js
```
import svelte from 'rollup-plugin-svelte-hot'
import Hmr from 'rollup-plugin-hot'
import resolve, { nodeResolve } from '@rollup/plugin-node-resolve'
import commonjs from '@rollup/plugin-commonjs'
import livereload from 'rollup-plugin-livereload'
import { terser } from 'rollup-plugin-terser'
import typescript from '@rollup/plugin-typescript'
import { copySync, removeSync } from 'fs-extra'
import { spassr } from 'spassr'
import autoPreprocess from 'svelte-preprocess'
import css from 'rollup-plugin-css-only'
import appConfig from './appConfig'

const { assetsDir, distDir, buildDir, template, script } = appConfig
const isNollup = !!process.env.NOLLUP
const production = !process.env.ROLLUP_WATCH
process.env.NODE_ENV = production ? 'production' : 'development'

// clear previous builds
removeSync(distDir)
removeSync(buildDir)

const serve = () => ({
    writeBundle: async () => {
        const options = {
            assetsDir: [assetsDir, distDir],
            entrypoint: template,
            script: script,
        }
        spassr({ ...options, port: 5002 })
        spassr({ ...options, ssr: true, port: 5005, ssrOptions: { inlineDynamicImports: true, dev: true } })
    },
})

const copyToDist = () => ({
    writeBundle() {
        copySync(assetsDir, distDir)
    },
})

export default {
    preserveEntrySignatures: false,
    input: ['src/main.ts'],
    output: {
        sourcemap: !production,
        format: 'esm',
        dir: buildDir,
        // for performance, disabling filename hashing in development
        chunkFileNames: `[name]${(production && '-[hash]') || ''}.js`,
    },
    plugins: [
        svelte({
            compilerOptions: {
                // enable run-time checks when not in production
                dev: !production,
            },
            emitCss: false,
            hot: isNollup,
            preprocess: [
                autoPreprocess({
                    postcss: require('./postcss.config.js'),
                    defaults: { style: 'postcss' },
                }),
            ],
        }),
        css({ output: `${distDir}/global.css` }),
        // resolve matching modules from current working directory
        resolve({
            module: false,
            browser: true,
            dedupe: (importee) => !!importee.match(/svelte(\/|$)/),
        }),
        nodeResolve(),
        commonjs(),
        typescript({
            sourceMap: !production,
            inlineSources: !production,
        }),

        production && terser(),
        !production && !isNollup && serve(),
        !production && !isNollup && livereload(distDir), // refresh entire window when code is updated
        !production && isNollup && Hmr({ inMemory: true, public: assetsDir }), // refresh only updated code
        {
            // provide node environment on the client
            transform: (code) => ({
                code: code.replace(/process\.env\.NODE_ENV/g, `"${process.env.NODE_ENV}"`),
                map: { mappings: '' },
            }),
        },
        production && copyToDist(),
    ],

    watch: {
        clearScreen: false,
        buildDelay: 100,
    },
}
```

package.json
```
{
   "name": "svelte-app",
   "version": "1.0.0",
   "@comments scripts": {
      "dev": "develop with blazing fast rebuilds",
      "dev:features": "develop with features like SSR and serviceworker enabled",
      "build": "run build scripts below",
      "build:app": "build single page application (SPA)",
      "build:static": "Generate static pages",
      "serve": "serve content in 'dist' folder",
      "rollup": "run the rollup bundler",
      "nollup": "run the nollup no-bundler",
      "routify": "run routify"
   },
   "scripts": {
      "dev": "run-p routify nollup",
      "dev:ssr": "run-p routify rollup",
      "build": "run-s build:*",
      "build:app": "routify -b && rollup -c",
      "build:static": "spank",
      "serve": "spassr --ssr",
      "rollup": "rollup -cw",
      "nollup": "nollup -c --verbose",
      "routify": "routify",
      "validate": "svelte-check"
   }
...
```

**Comment by PepsRyuu** [comment]
The later versions of ```@rollup/plugin-commonjs``` rely on live bindings. This isn't enabled by default due to the severe performance penalty it introduces. You can enable live bindings using the experimental ``--live-bindings``` flag (or ```liveBindings: true```). 

Alternatively, you can swap out ```@rollup/plugin-commonjs``` with my ```rollup-plugin-commonjs-alternate``` which will work just fine without the need for enabling live bindings.

**Comment by risalfajar** [comment]
Thank you, with `rollup-plugin-commonjs-alternate` it works now.

**Comment by ferulisses** [comment]
Took me a full day to diagnose and finally get here, just leaving this comment to anyone googling.

I got the error for any Javascript that I tried to import:
`import jquery from 'jquery'`

Got the error:
`TypeError: jquery is not a function`

When using:
`console.log(jquery);` 

I got the result:

`
{}
[[Prototype]]: Object
`

The possible solutions:
1. Swap @rollup/plugin-commonjs with rollup-plugin-commonjs-alternate as suggested 
  - Install the package: 
    npm install rollup-plugin-commonjs-alternate
  - At rollup.js, replace the line: 
    import commonjs from '@rollup/plugin-commonjs';
  - With:
    import commonjs from 'rollup-plugin-commonjs-alternate';
2. Downgrade @rollup/plugin-commonjs to 18.1.0 (before they started to use live bindings)
3. Enable Live Bindings:
  https://github.com/PepsRyuu/nollup/blob/master/docs/live-bindings.md
  - Edit .nolluprc.js, add the line:
    liveBindings: true

Restart npm after any of the solutions.

I'm not sure wich one is the best, probably the number 2 will be the worst when updating components.
